lib/compat/wordpress-6.5/blocks.php

```diff
@@ -123,6 +123,16 @@ function gutenberg_block_bindings_replace_html( $block_content, $block_name, str
 							}
 							return $amended_button->get_updated_html();
 						}
+						if ( 'core/image' === $block_name && 'caption' === $attribute_name ) {
+							// TODO: Don't use regex.
+							return preg_replace_callback(
+								'/<figcaption[^>]*>.*?<\/figcaption>/is',
+								function () use ( $amended_content ) {
+									return $amended_content->get_updated_html();
+								},
+								$block_content
+							);
```

Comment:
I don't think there's a need to use a callback:
```suggestion
							return preg_replace(
								'/<figcaption[^>]*>.*?<\/figcaption>/is',
								$amended_content->get_updated_html(),
								$block_content
							);
```